diff --git a/dist/emitter.js b/dist/emitter.js
index ea7c9c50bc9d7e6e9fb017e03a89eb2613529c10..adbd00985ed2575d5712d0cc8d7adc6455176726 100644
--- a/dist/emitter.js
+++ b/dist/emitter.js
@@ -16,6 +16,11 @@ export const intrinsicNameToTSType = new Map([
     ["null", "null"],
     ["never", "never"],
     ["void", "void"],
+    ["utcDateTime", "Date"],
+    ["offsetDateTime", "Date"],
+    ["plainDate", "Date"],
+    ["plainTime", "Date"],
+    ["duration", "string"],
 ]);
 export class TypescriptEmitter extends CodeTypeEmitter {
     nsByName = new Map();
